src/pocketmine/network/mcpe/NetworkBinaryStream.php

```diff
@@ -74,6 +77,62 @@ public function putUUID(UUID $uuid) : void{
 		$this->putLInt($uuid->getPart(2));
 	}
 
+	public function getSkin() : Skin{
+		$skinId = $this->getString();
+		$skinResourcePatch = $this->getString();
+		$skinData = $this->getImage();
+		$animationCount = $this->getLInt();
+		$animations = [];
+		for($i = 0; $i < $animationCount; ++$i){
+			$animations[] = new SkinAnimation($this->getImage(), $this->getLInt(), $this->getLFloat());
```

Comment:
For the sake of code clarity, please make the following change:
```suggestion
			$animations[] = new SkinAnimation(
				$image = $this->getImage(),
				$animationType = $this->getLInt(),
				$animationFrames = $this->getLFloat()
			);
```
This makes it clear what the purpose of the fields being read are, and refactoring will persist these assignments.